lib/BuildSystem/BuildSystem.cpp

```diff
@@ -2853,12 +2853,30 @@ class SwiftCompilerShellCommand : public ExternalCommand {
     return ExternalCommand::configureAttribute(ctx, name, values);
   }
 
+  template <typename String>
+  std::string escapeForJson(String const& in) const {
+    std::ostringstream ss;
+    for (auto i = in.begin(); i != in.end(); i++) {
+      switch (*i) {
```

Comment:
Only escape characters less than 0x20, backslash, and quote, otherwise add to the output literally.